### DISCUSSION TITLE
Run Version 2 on Raspberry Pi and changed time interval

### DISCUSSION THREAD

**Discussion by nymJ** [danielfernau/unifi-protect-video-downloader#154] [Ideas]
Hello,
First of all, thank you for this super tool. I used version 1 in a slightly modified way in the Smart Home context. When someone rang the doorbell, I exported the last 10 seconds before the bell rang and sent a message to my smartphone via Telegram Messenger. The whole thing runs on a Raspberry Pi 4. After updating my cloud key, version 1 can no longer be used (error message 404). Now I've tried your current version. Unfortunately I can't get the Docker container installed on the Raspberry (no ARM64) under Ubuntu it works. In addition, I lack the knowledge to adapt the container so that it only exports the last 10 seconds again. Do you have any ideas or suggestions that could help me to implement them? Thank you

**Comment by danielfernau** [comment]
Hey there!

Yes, you've probably updated to the newer firmwares which are "UniFi OS", and by that the paths and ports to access the Protect API change. Using the most recent version you should be able to access it again.

Right now there's an arm64 image -- let me check if I can add a 32-bit one as well... I'll let you know shortly.

Another issue you might run into (but as far as I remember that part is identical to the v1.x code) is that the code doesn't really allow you to download just 10 seconds right now. Someone else also opened an issue about that a while back and was able to make it fit their needs: https://github.com/danielfernau/unifi-protect-video-downloader/issues/57
However, I didn't find the time to make the necessary changes yet; but it's still planned. I'll add a flag to bypass the alignment part.

If you need any additional help or details just let me know and I'll see what I can do!

**Reply by nymJ** [reply]
Hi,

thank you much for your reply. Which ports and paths I have to change? port 443 or 7443? And which path do you mean?

**Reply by danielfernau** [reply]
The old variant looks like this: https://github.com/danielfernau/unifi-protect-video-downloader/blob/master/protect_archiver/client/legacy.py#L30-L31
Whereas the new API looks like this: https://github.com/danielfernau/unifi-protect-video-downloader/blob/master/protect_archiver/client/unifi_os.py#L30-L31
Basically, instead of port 7443 and `/api` you now need to use port 443 and `/proxy/protect/api`. The commands, however, are mostly the same. If you want to try and backport the new API structure into the v1.x code, have a look at the two files linked above. They contain all the necessary paths and ports.

Adding a 32bit ARM image is a bit more challenging than I initially thought; I need to solve a problem with installing Python Poetry's dependencies first. Will post an update here once it works and is available on Docker Hub.

**Reply by danielfernau** [reply]
... was able to add arm32v7 builds by manually setting the platform to amd64 for the first build stage in the Dockerfile.

The `edge` channel now has an ARMv7 image available:
`docker pull unifitoolbox/protect-archiver:edge` 

PR https://github.com/danielfernau/unifi-protect-video-downloader/pull/162
